web/pdf_thumbnail_viewer.js

```diff
@@ -30,201 +30,193 @@ var THUMBNAIL_SCROLL_MARGIN = -19;
-      }
-      var visibleThumbs = this._getVisibleThumbs();
-      var numVisibleThumbs = visibleThumbs.views.length;
-
-      // If the thumbnail isn't currently visible, scroll it into view.
-      if (numVisibleThumbs > 0) {
-        var first = visibleThumbs.first.id;
-        // Account for only one thumbnail being visible.
-        var last = (numVisibleThumbs > 1 ? visibleThumbs.last.id : first);
-        if (page <= first || page >= last) {
-          scrollIntoView(thumbnail, { top: THUMBNAIL_SCROLL_MARGIN, });
-        }
-      }
-    },
+  /**
+   * @private
+   */
+  _scrollUpdated() {
+    this.renderingQueue.renderHighestPriority();
+  }
+
+  getThumbnail(index) {
+    return this.thumbnails[index];
+  }
 
-    get pagesRotation() {
-      return this._pagesRotation;
-    },
+  /**
+   * @private
+   */
+  _getVisibleThumbs() {
+    return getVisibleElements(this.container, this.thumbnails);
+  }
 
-    set pagesRotation(rotation) {
-      this._pagesRotation = rotation;
-      for (var i = 0, l = this.thumbnails.length; i < l; i++) {
-        var thumb = this.thumbnails[i];
-        thumb.update(rotation);
+  scrollThumbnailIntoView(page) {
+    let selected = document.querySelector('.thumbnail.selected');
+    if (selected) {
+      selected.classList.remove('selected');
+    }
+    let thumbnail = document.querySelector(
+      'div.thumbnail[data-page-number="' + page + '"]');
+    if (thumbnail) {
+      thumbnail.classList.add('selected');
+    }
+    let visibleThumbs = this._getVisibleThumbs();
+    let numVisibleThumbs = visibleThumbs.views.length;
+
+    // If the thumbnail isn't currently visible, scroll it into view.
+    if (numVisibleThumbs > 0) {
+      let first = visibleThumbs.first.id;
+      // Account for only one thumbnail being visible.
+      let last = (numVisibleThumbs > 1 ? visibleThumbs.last.id : first);
+      if (page <= first || page >= last) {
+        scrollIntoView(thumbnail, { top: THUMBNAIL_SCROLL_MARGIN, });
       }
-    },
+    }
+  }
 
-    cleanup: function PDFThumbnailViewer_cleanup() {
-      PDFThumbnailView.cleanup();
-    },
+  get pagesRotation() {
+    return this._pagesRotation;
+  }
 
-    /**
-     * @private
-     */
-    _resetView: function PDFThumbnailViewer_resetView() {
-      this.thumbnails = [];
-      this._pageLabels = null;
-      this._pagesRotation = 0;
-      this._pagesRequests = [];
+  set pagesRotation(rotation) {
+    this._pagesRotation = rotation;
+    for (let i = 0, l = this.thumbnails.length; i < l; i++) {
+      this.thumbnails[i].update(rotation);
+    }
+  }
 
-      // Remove the thumbnails from the DOM.
-      this.container.textContent = '';
-    },
+  cleanup() {
+    PDFThumbnailView.cleanup();
+  }
 
-    setDocument: function PDFThumbnailViewer_setDocument(pdfDocument) {
-      if (this.pdfDocument) {
-        this._cancelRendering();
-        this._resetView();
-      }
+  /**
+   * @private
+   */
+  _resetView() {
+    this.thumbnails = [];
```

Comment:
Since you're touching this code anyway, mind changing it to  @`this._thumbnails` instead throughout the file?
This would address an inconsistency between `PDFViewer`/`PDFThumbnailViewer` (where the former uses `this._pages`), since `this.getThumbnail` should be used from outside of the class anyway.